### ISSUE TITLE
Solution import with Canvas App fails with error 'CanvasApp import: FAILURE: No file with that name found

### ISSUE THREAD

**Issue by parvezghumra** [microsoft/powerplatform-build-tools#684]
We're using the Power Platform Github Actions for our ALM processes, which I understand are wrapper around the PAC CLI.

**The 'import-solution' action is failing currently with the following error:**

`The reason given was: CanvasApp import: FAILURE: No file with that name found
  Microsoft PowerPlatform CLI
  Version: 1.28.3+g4ba7dfd`

**The command which fails via Github Actions is:**

`Calling pac cli inputs: solution import --path D:\a\....solution_managed.zip --async true --force-overwrite true --publish-changes false --skip-dependency-check false --convert-to-managed false --max-async-wait-time 60 --activate-plugins true --skip-lower-version false --import-as-holding true --settings-file .....\deploymentsettings.json`

**Our commit process involves invoking the following commands via Github Actions (all of which succeed and do not unpack Canvas Apps)**

`Calling pac cli inputs: solution export --name solution --path D:....\solution.zip --overwrite true --managed false --async true --max-async-wait-time 60`

`Calling pac cli inputs: solution export --name solution --path D:\.....\solution_managed.zip --overwrite true --managed true --async true --max-async-wait-time 60`

`Calling pac cli inputs: solution unpack --zipFile D:\....\solution.zip --folder D:.....\solution --packageType Both --errorlevel Info --singleComponent None --useLcid false --useUnmanagedFileForMissingManaged false --disablePluginRemap false --allowDelete true --allowWrite true --clobber true`

`Calling pac cli inputs: solution unpack --zipFile D:\.....\solution.zip --folder D:\.....\solution --packageType Both --errorlevel Info --singleComponent None --useLcid false --useUnmanagedFileForMissingManaged false --disablePluginRemap false --allowDelete true --allowWrite true --clobber true`

**Our build process involves invoking the following commands via Github Actions (all of which succeed and do not repack Canvas Apps)**

`Calling pac cli inputs: solution pack --zipFile /home/runner/work/....../solution.zip --folder /home/runner/work/...../solution --packageType Both --errorlevel Info --singleComponent None --useLcid false --useUnmanagedFileForMissingManaged false --disablePluginRemap false`

All of this setup was working fine until yesterday

**Comment by parvezghumra** [comment]
Looking at this response on a [similar issue on the Community forum](https://powerusers.microsoft.com/t5/Building-Power-Apps/Canvas-App-Release-Pipeline-Failing-with-Error-No-file-with-that/m-p/2506698/highlight/true#M623043), it looks like someone resolved this by using version 2 of the corresponding Azure DevOps Power Platform Build Tools. I've verified that I'm using the latest available version of the corresponding Github Action but this hasn't resolved the issue.

**Comment by bananabrann** [comment]
@parvezghumra This is the issue our team is having too; I've been talking in #677. Seems similar enough 🤷‍♂️ I haven't confirmed 2.0.47 works because of our bureaucratic processes to update the build tools from 2.0.5 on ADO Marketplace

Currently our workaround is to import manually across our environments. It's a pain, but the world is punishing me for taking other technology for granted 😝

**Comment by parvezghumra** [comment]
Thanks @bananabrann. I guess the obvious difference is that I'm getting this error on the import solution operation but you're experiencing on the unpack/pack operation

Also I'm using GitHub Actions and the latest version but still facing this.

I'll keep an eye on the other issue. Thanks 👍

**Comment by parvezghumra** [comment]
Thanks for following up @bananabrann for following up.

I have verified that a manually exported solution from the source development environment successfully imports manually into the target environment. And as you have highlighted, even exporting the solution from the source environment using an automated process and then importing this to the target succeeds. The failure occurs if you unpack the exported solution, repack it and then attempt to import this repacked version to the target environment.

The challenge I have is that our current setup is designed to be source control centric with an emphasis on end to end traceability as this gives us confidence over the versions of artifacts that are deployed. Therefore, we do not hold the exported solution anywhere and rely on our build process to generate the solution to be deployed.

I may have to adopt an approach similar to what you describe in your workaround for now, but I am still waiting on Microsoft to come back to me on this.

**Comment by RobertMadar** [comment]
Hello, have similiar problem with Import part. Do you have any news? 

Solution Importing...
  Microsoft PowerPlatform CLI
  Version: 1.25.2+g1c4a1b8
  Online documentation: https://aka.ms/PowerPlatformCLI
  Feedback, Suggestions, Issues: https://github.com/microsoft/powerplatform-build-tools/discussions
  
  Error: CanvasApp import: FAILURE: No file with that name found
      CanvasApp import: FAILURE: No file with that name found

**Comment by parvezghumra** [comment]
@RobertMadar As this doesn't appear to be a platform issue, my official support request was closed as that is not the right channel. I'm told via other channels that the appropriate MS team (I'm guessing the team that looks after the PAC CLI, PPBT and/or Github Actions) is looking at this. I'll follow up next week.

**Comment by fcctsorrell** [comment]
I'm having the same problem.  Any update???  Thanks.

**Comment by snizar007** [comment]
We are investigating this issue.

**Comment by parvezghumra** [comment]
Hi @snizar007 ,

Thanks for your time on the call yesterday. I have done a bit of testing by running the `pac solution unpack`, `pac solution pack` and `pac solution import` commands locally on my machine using various versions of the PAC CLI.

It seems that the latest version of the Power Platform Github Actions (version 1.2.0) is still using version 1.28.3 of the PAC CLI, and so the unpack operation does not generate the **/solution/CanvasApps/<CanvasAppName>_AdditionalUris0_identity.json** file mentioned in the community thread I referenced in my earlier comment. I was able to replicate this same behaviour using this version of the PAC CLI running it locally on my machine. But using any later version of the PAC CLI (version 1.29.6 onwards), the unpack operation successfully generates this file, the pack operation successfully builds the unmanaged and managed solutions and the import operation successfully imports the resulting managed solution into the target environment.

So in summary, we need a new release of the Power Platform Github Actions which use version 1.29.6 of the PAC CLI or later. :) When do you expect to publish this please?

Thanks,
Parvez

**Comment by snizar007** [comment]
I have been able to reproduce this issue and our engineering team is working on the fix. In the meantime, you can force PAC CLI ver: 1.29.11 by using below:
![image](https://github.com/microsoft/powerplatform-build-tools/assets/116691001/14a3123d-337d-4964-a267-563cbb8a2c91)

**Comment by parvezghumra** [comment]
Thanks @snizar007. Just to confirm that your suggestion has resolved the issue for us for now. We are using it as a temporary workaround until a new version of the Github Actions are available  with the permanent fix.

**Comment by snizar007** [comment]
Thank you @parvezghumra for confirming.

**Comment by talyrobinonsemi** [comment]
> force PAC CLI ver: 1.29.11

I am having the same issue.  @snizar007, may I ask where did you update the version - how did you get to this logic? I am using the GUI to build the pipeline tasks, so unsure of where exactly to update the PAC CLI version. Any advice would be greatly appreciated.

**Comment by greggdavenport** [comment]
Sorry to say this did not work for me. Any other solutions would be much appreciated.

**Comment by StefDeBisschop** [comment]
Any update on this issue?

**Comment by greggdavenport** [comment]
@StefDeBisschop , in the [community forum mentioned above](https://powerusers.microsoft.com/t5/Building-Power-Apps/Canvas-App-Release-Pipeline-Failing-with-Error-No-file-with-that/m-p/2506698/highlight/true#M623043), some members said deleting the <additionalUri> node from the canvas apps metadata XML was a good workaround. I have also found this to be the only solution that works for me, and I’ve used this in several repos.

**Comment by StefDeBisschop** [comment]
@greggdavenport The solution provided works as a workaround! Hope Microsoft will patch this sooner than later

**Comment by fcctsorrell** [comment]
Can we expect an update on this any time soon?

**Comment by petrochuk** [comment]
> @greggdavenport The solution provided works as a workaround! Hope Microsoft will patch this sooner than later

Nuget library used for canvas pack/unpack is available here: https://github.com/microsoft/PowerApps-Tooling. There is an effort to change the approach which will eliminate the need to do canvas pack/unpack.

**Comment by fcctsorrell** [comment]
> > @greggdavenport The solution provided works as a workaround! Hope Microsoft will patch this sooner than later
> 
> Nuget library used for canvas pack/unpack is available here: https://github.com/microsoft/PowerApps-Tooling. There is an effort to change the approach which will eliminate the need to do canvas pack/unpack.

Any update?  Any idea when this new approach may be implemented?

**Comment by petrochuk** [comment]
> > > @greggdavenport The solution provided works as a workaround! Hope Microsoft will patch this sooner than later
> > 
> > 
> > Nuget library used for canvas pack/unpack is available here: https://github.com/microsoft/PowerApps-Tooling. There is an effort to change the approach which will eliminate the need to do canvas pack/unpack.
> 
> Any update? Any idea when this new approach may be implemented?

It is in public preview now: https://www.microsoft.com/en-us/power-platform/blog/power-apps/introducing-code-view-in-power-apps-studio-public-preview/

**Comment by greggdavenport** [comment]
> > > > @greggdavenport The solution provided works as a workaround! Hope Microsoft will patch this sooner than later
> > > 
> > > 
> > > Nuget library used for canvas pack/unpack is available here: https://github.com/microsoft/PowerApps-Tooling. There is an effort to change the approach which will eliminate the need to do canvas pack/unpack.
> > 
> > 
> > Any update? Any idea when this new approach may be implemented?
> 
> It is in public preview now: https://www.microsoft.com/en-us/power-platform/blog/power-apps/introducing-code-view-in-power-apps-studio-public-preview/

I saw in [this video](http://www.youtube.com/watch?v=jH0QE9bQz5U) from Build 2024 that the team is seeking feedback on what other source control should be supported besides Azure DevOps. Where can we best leave feedback to advocate for Github?

**Comment by fcctsorrell** [comment]
> > > > > @greggdavenport The solution provided works as a workaround! Hope Microsoft will patch this sooner than later
> > > > 
> > > > 
> > > > Nuget library used for canvas pack/unpack is available here: https://github.com/microsoft/PowerApps-Tooling. There is an effort to change the approach which will eliminate the need to do canvas pack/unpack.
> > > 
> > > 
> > > Any update? Any idea when this new approach may be implemented?
> > 
> > 
> > It is in public preview now: https://www.microsoft.com/en-us/power-platform/blog/power-apps/introducing-code-view-in-power-apps-studio-public-preview/
> 
> I saw in [this video](http://www.youtube.com/watch?v=jH0QE9bQz5U) from Build 2024 that the team is seeking feedback on what other source control should be supported besides Azure DevOps. Where can we best leave feedback to advocate for Github?

I would like to know this as well.  2 years ago when I began to develop in Power Apps and had to decide for our company whether to pursue Devops or Github, I went with Github because it was the future of MS source control, I thought...

**Comment by petrochuk** [comment]
You can join [Community Office Hours](https://github.com/microsoft/powerplatform-build-tools/blob/main/community/Office_Hours.md) later this week and share it there. I think some members of the team that works on source control integration might be there.

**Comment by fcctsorrell** [comment]
> You can join [Community Office Hours](https://github.com/microsoft/powerplatform-build-tools/blob/main/community/Office_Hours.md) later this week and share it there. I think some members of the team that works on source control integration might be there.

Has there been any discussion of this in the Community Office Hours?  I haven't been able to attend.  Any other update on this issue?

**Comment by yonass08** [comment]
I had a solution that contains a command library for a model driven app. When I export and unpack the solution the additionalUri_identity.json is missing for the canvas app and whenever I try to deploy it using the deployment packager I get this error "PackageDeployer Information: 8 : Message: errorDetails - Message  - CanvasApp import: FAILURE: The solution specified an expected assets file but that file was missing or invalid.
"

**Comment by StefDeBisschop** [comment]
@greggdavenport @fcctsorrell @yonass08 Since i've created a V2 for my project that used this approach before, i've been using XrmToolbox with Solution Transfer Tool to transfer and build my solution to the production enviroment; works like a charm! Not going back from this :) Hope this helps!
